modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/StrategyElement/ServiceRequest/FhirServiceRequestMapping.php

```diff
@@ -0,0 +1,370 @@
+            $FHIRServiceRequest->getEncounter()->getReference()->setValue($refEncounter);
+        }
+
+        $FHIRServiceRequest->getReasonCode()[0]->getCoding()[0]->setCode($ServiceRequestFromDb['reason_code']);
+
+        if(!is_null($ServiceRequestFromDb['patient'])){
+            $refSubject = self::PATIENT_URI . $ServiceRequestFromDb['patient'];
+            $FHIRServiceRequest->getSubject()->getReference()->setValue($refSubject);
+        }
+
+
+        if (!is_null($ServiceRequestFromDb['instruction_code'])) {
+            $codeCode = $FHIRServiceRequest->getCode();
+            $codeCoding = $codeCode->getCoding()[0];
+
+            $title=$this->getValueSetTitle(self::CODE_SYSTEM,$ServiceRequestFromDb['instruction_code']);
+            $codeCode->getText()->setValue($title);
+            $codeCoding->getCode()->setValue($ServiceRequestFromDb['instruction_code']);
+            $codeCoding->getSystem()->setValue(self::LIST_SYSTEM_LINK . self::CODE_SYSTEM);
+        }
+
+        $orderDetailCoding =$FHIRServiceRequest->getOrderDetail()[0];
+
+        if(!is_null($ServiceRequestFromDb['order_detail_code'])  && !is_null($ServiceRequestFromDb['order_detail_system']) ){
+            $orderDetail = $orderDetailCoding->getCoding()[0];
+            $title=$this->getValueSetTitle($ServiceRequestFromDb['order_detail_system'],$ServiceRequestFromDb['order_detail_code']);
+            $orderDetailCoding->getText()->setValue($title);
+            $orderDetail->setCode($ServiceRequestFromDb['order_detail_code']);
+            $orderDetail->getSystem()->setValue(self::LIST_SYSTEM_LINK . $ServiceRequestFromDb['order_detail_system']);
+        }
+
+        $FHIRServiceRequest->getPatientInstruction()->setValue($ServiceRequestFromDb['patient_instruction']);
+
+        if(!is_null($ServiceRequestFromDb['requester'])){
+            $refReference = self::PRACTITIONER_URI . $ServiceRequestFromDb['requester'];
+            $FHIRServiceRequest->getRequester()->getReference()->setValue($refReference);
+        }
+
+        if(!is_null($ServiceRequestFromDb['performer'])){
+            $refPerformer = self::PRACTITIONER_URI . $ServiceRequestFromDb['performer'];
+            $FHIRServiceRequest->getPerformer()[0]->getReference()->setValue($refPerformer);
+        }
+
+        if(!is_null($ServiceRequestFromDb['reason_reference_doc_id'])){
+            $refReasonReference = self::DOCUMENT_REFERENCE_URI . $ServiceRequestFromDb['reason_reference_doc_id'];
+            $FHIRServiceRequest->getReasonReference()[0]->getReference()->setValue($refReasonReference);
+        }
+
+        if($ServiceRequestFromDb['occurrence_datetime']!==self::EMPTY_TIME){
+            $occurrenceDateTime = $this->createFHIRDateTime(null, null, $ServiceRequestFromDb['occurrence_datetime']);
+            $FHIRServiceRequest->setOccurrenceDateTime($occurrenceDateTime);
+        }
+
+        if($ServiceRequestFromDb['authored_on']!==self::EMPTY_TIME){
```

Comment:
I see the authored_on column is default null, so when do it can fill with '00-00-00 00:00' ?